### PR TITLE
Fix StatsD drop metrics tags when using summary as observer_type for timer/histogram

### PR DESCRIPTION
**Pull request by gavindoudou**
**Description:** <Describe what has changed.>

- When sending timer/histogram using OTLP summary types, StatsD receiver does not send metrics tags. Add metrics tags when creating OTLP summary. 

- Deprecate `Append` with `AppendEmpty`

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3431

**Testing:** <Describe what testing was performed and which tests were added.>
Added the test part for tags in creating OTLP summary metrics.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by jmacd — APPROVED** [review]

File: receiver/statsdreceiver/README.md [hunk 1]
```diff
@@ -6,8 +6,6 @@ Supported pipeline types: metrics
 
 Use case: it does not support horizontal pool of collectors. Desired work case is that customers use the receiver as an agent with a single input at the same time.
 
-> :construction: This receiver is currently in **BETA**.
```
Comment (bogdandrutu) [inline receiver/statsdreceiver/README.md:9]: Please split this in a separate PR

**Review by bogdandrutu — DISMISSED** [review]

File: receiver/statsdreceiver/protocol/statsd_parser.go [hunk 2]
```diff
@@ -108,15 +108,21 @@ func (p *StatsDParser) GetMetrics() pdata.Metrics {
 	rm := metrics.ResourceMetrics().AppendEmpty()
 
 	for _, metric := range p.gauges {
-		rm.InstrumentationLibraryMetrics().Append(metric)
+		ilm := rm.InstrumentationLibraryMetrics().AppendEmpty()
+		metric.CopyTo(ilm)
+		//rm.InstrumentationLibraryMetrics().Append(metric)
```
Comment (Aneurysm9) [inline receiver/statsdreceiver/protocol/statsd_parser.go:113]:
```suggestion
```

File: receiver/statsdreceiver/protocol/statsd_parser.go [hunk 3]
```diff
@@ -108,15 +108,21 @@ func (p *StatsDParser) GetMetrics() pdata.Metrics {
...
 	}
 
 	for _, metric := range p.counters {
-		rm.InstrumentationLibraryMetrics().Append(metric)
+		ilm := rm.InstrumentationLibraryMetrics().AppendEmpty()
+		metric.CopyTo(ilm)
+		//rm.InstrumentationLibraryMetrics().Append(metric)
```
Comment (Aneurysm9) [inline receiver/statsdreceiver/protocol/statsd_parser.go:119]:
```suggestion
```

File: receiver/statsdreceiver/protocol/statsd_parser.go [hunk 4]
```diff
@@ -108,15 +108,21 @@ func (p *StatsDParser) GetMetrics() pdata.Metrics {
...
 	}
 
 	for _, metric := range p.timersAndDistributions {
-		rm.InstrumentationLibraryMetrics().Append(metric)
+		ilm := rm.InstrumentationLibraryMetrics().AppendEmpty()
+		metric.CopyTo(ilm)
+		//rm.InstrumentationLibraryMetrics().Append(metric)
```
Comment (Aneurysm9) [inline receiver/statsdreceiver/protocol/statsd_parser.go:125]:
```suggestion
```

**Review by Aneurysm9 — APPROVED** [review]

File: receiver/statsdreceiver/README.md (hunk 1 above)
Comment (gavindoudou) [reply receiver/statsdreceiver/README.md]: separated

File: receiver/statsdreceiver/protocol/statsd_parser.go [hunk 5]
```diff
@@ -108,15 +108,18 @@ func (p *StatsDParser) GetMetrics() pdata.Metrics {
...
 	for _, metric := range p.gauges {
-		rm.InstrumentationLibraryMetrics().Append(metric)
+		ilm := rm.InstrumentationLibraryMetrics().AppendEmpty()
+		metric.CopyTo(ilm)
 	}
 
 	for _, metric := range p.counters {
-		rm.InstrumentationLibraryMetrics().Append(metric)
+		ilm := rm.InstrumentationLibraryMetrics().AppendEmpty()
+		metric.CopyTo(ilm)
```
Comment (bogdandrutu) [inline receiver/statsdreceiver/protocol/statsd_parser.go:117]: Inline:

`metric.CopyTo(rm.InstrumentationLibraryMetrics().AppendEmpty())`

**Comment by gavindoudou** [comment]
Can we merge it today? It's a bugfix. Thanks!
